galasa-managers-parent/galasa-managers-workflow-parent/dev.galasa.githubissue.manager/src/main/java/dev/galasa/githubissue/internal/GitHubIssueManagerImpl.java

```diff
@@ -3,61 +3,288 @@
+	
 	private GitHubIssue gitHubIssue;
 	
+	private IHttpManagerSpi  httpManager;
+	private IHttpClient      httpClient;
+	
 	@Override
 	public void initialise(@NotNull IFramework framework, @NotNull List<IManager> allManagers,
 			@NotNull List<IManager> activeManagers, @NotNull GalasaTest galasaTest) throws ManagerException {
 		super.initialise(framework, allManagers, activeManagers, galasaTest);
-		logger.info("In the GitHubIssueManager");
-		// Check to see if GitHubIssue annotation is present in the test class
-		// If it is, need to activate
+		
+		// Check to see if GitHubIssue annotation is present in the test class or on any of the test methods
+		// If it is, activate this manager
 		
 		if (!required) {
+			
 			if (galasaTest.isJava()) {
 				Class<?> testClass = galasaTest.getJavaTestClass();
 				
-				this.gitHubIssue = testClass.getAnnotation(GitHubIssue.class);
-				if (this.gitHubIssue == null) {
-					return; // Not required
+				this.gitHubIssue = testClass.getAnnotation(GitHubIssue.class); // Set globally as might need later
+				if (this.gitHubIssue == null) { 
+					
+					GitHubIssue gitHubIssue = null;
+					for (Method testMethod : testClass.getMethods()) {
+						gitHubIssue = testMethod.getAnnotation(GitHubIssue.class);
+						if (gitHubIssue != null) {
+							required = true;
+							break;
+						}
+					}
+					// Annotation not present on the class or any methods
+					if (!required) {
+						return;
+					}
 				}
 			} else {
 				return; // Only support Java
 			}
 		}
 		
-		logger.info("Annotation found");
+		logger.info("GitHubIssueManager needed");
 		
+		this.galasaTest = galasaTest;
+		
+		try {
+            GitHubIssuePropertiesSingleton.setCps(framework.getConfigurationPropertyService(NAMESPACE));
+        } catch (ConfigurationPropertyStoreException e) {
+            throw new GitHubIssueManagerException("Unable to request framework services", e);
+        }	
 		
 		youAreRequired(allManagers, activeManagers, galasaTest);
 	
 	}
 	
 	@Override
-	public void youAreRequired(@NotNull List<IManager> allManagers, @NotNull List<IManager> activeManagers, @NotNull GalasaTest galasaTest) {
+	public void youAreRequired(@NotNull List<IManager> allManagers, @NotNull List<IManager> activeManagers, @NotNull GalasaTest galasaTest) throws ManagerException {
+		super.youAreRequired(allManagers, activeManagers, galasaTest);
+		
+		if (activeManagers.contains(this)) {
+			return;
+		}
+		
+		this.required = true;
+		activeManagers.add(this);	
+		
+		this.httpManager = addDependentManager(allManagers, activeManagers, galasaTest, IHttpManagerSpi.class);
+        if (this.httpManager == null) {
+            throw new GitHubIssueManagerException("Unable to locate the Http Manager, required for the GitHub Issue Manager");
+        }
+	}
+	
+	@Override
+    public boolean areYouProvisionalDependentOn(@NotNull IManager otherManager) {
+        if (otherManager instanceof IHttpManager) {
+            return true;
+        }
```

Comment:
although we use HTTPManager I don't think were are dependent on it.  HTTP Manager for the SPI doesn't provision anything so we are probably OK